File: package/umt_python/tests/unit/string/test_deburr.py
import unittest

from src.string import deburr


class TestDeburr(unittest.TestCase):
    def test_removes_acute_accents(self):
        self.assertEqual(deburr("déjà vu"), "deja vu")

    def test_removes_multiple_marks(self):
        self.assertEqual(deburr("Crème brûlée"), "Creme brulee")

    def test_mix_of_accented_and_plain(self):
        self.assertEqual(deburr("résumé café"), "resume cafe")

    def test_plain_ascii_unchanged(self):
        self.assertEqual(deburr("hello world"), "hello world")

    def test_empty_string(self):
        self.assertEqual(deburr(""), "")


if __name__ == "__main__":
    unittest.main()

File: package/umt_rust/tests/string/test_deburr.rs
use umt_rust::string::umt_deburr;

#[test]
fn test_removes_acute_accents() {
    assert_eq!(umt_deburr("déjà vu"), "deja vu");
}

#[test]
fn test_removes_multiple_diacritical_marks() {
    assert_eq!(umt_deburr("Crème brûlée"), "Creme brulee");
}

#[test]
fn test_handles_mix_of_accented_and_plain() {
    assert_eq!(umt_deburr("résumé café"), "resume cafe");
}

#[test]
fn test_leaves_plain_ascii_unchanged() {
    assert_eq!(umt_deburr("hello world"), "hello world");
}

#[test]
fn test_handles_empty_string() {
    assert_eq!(umt_deburr(""), "");
}
